### PR TITLE
bug: could sometimes fail to export all repeat rows. depends on chunks.

### PR DESCRIPTION
**Pull request by issa-tseng**
* this can be a hard bug to reproduce. the number of bytes of csv
  produced per row of data, as well as the size of the rows coming from
  the database.
* whatever the conditions, the issue is that for whatever reason the csv
  stringify stream doesn't always wait to flush out, so it erroneously
  reports to the zip archive that it's done.
* if we wrap it in a PassThrough it works perfectly.
* there is probably a different solution that involves patching
  csv-stringify, but that's a far more expensive operation than we have
  time for rn.

### CODE REVIEW

**Comment by issa-tseng** [comment]
fwiw the unit test never repro'd the issue. it took the whole service, as well as somewhat exactly the test data you gave me for it to occur.

**Review by lognaturel — CHANGES_REQUESTED** [review]
We’ve talked about this privately but want to make sure the PR status is clear in case anyone else takes a look.

The unit test can be made to fail before code changes by adding 4 child repeat instances per submission. With the code change here, it fails with 12 child repeat instances or 526+ of the original submissions (single repeat instance).

File: test/unit/data/briefcase.js [hunk 1]
```diff
@@ -352,6 +353,59 @@ Candace,2,three,three/children/child[1]
...
+              <input ref="/data/children/child/name">
+                <label>What is the child's name?</label>
+              </input>
+            </repeat>
+          </group>
+        </h:body>
+      </h:html>`;
+
+    const inStream = streamTest.fromObjects(
+      (new Array(50)).fill(null).map(_ => instance(uuid(), `<orx:meta><orx:instanceID>${uuid()}</orx:instanceID></orx:meta><name>${uuid()}</name><children><child><name>${uuid()}</name></child></children>`)));
```
Comment (lognaturel) [inline test/unit/data/briefcase.js:398]: This test passes without the change. Is that deliberate? It's unlikely that we'd have the exactly same regression so maybe it's ok. I'd tend to do 127 submissions, though, which does fail.

File: test/unit/data/briefcase.js [hunk 2]
```diff
@@ -352,6 +353,59 @@ Candace,2,three,three/children/child[1]
...
+
+    callAndParse(inStream, formXml, 'singlerepeat', (result) => {
+      result.filenames.should.containDeep([ 'singlerepeat.csv', 'singlerepeat-child.csv' ]);
+      result['singlerepeat.csv'].split('\n').length.should.equal(52);
+      result['singlerepeat-child.csv'].split('\n').length.should.equal(52);
+      console.log('aoeuaoeu');
```
Comment (lognaturel) [inline test/unit/data/briefcase.js:404]: Remove?

File: test/unit/data/briefcase.js (hunk 1 above)
Comment (issa-tseng) [reply test/unit/data/briefcase.js:398]: yeah i didn't think it mattered but if 127 verifies the change let's do it.

**Review by lognaturel — APPROVED** [review]

**Review by matthew-white — APPROVED** [review]
I looked over 3889ee3ff2f3c9370907622cee51721cb1894717, which seemed reasonable. :+1:

**Pull request merged**